includes/class-plugin-manager.php

```diff
@@ -0,0 +1,302 @@
+<?php
+/**
+ * Newspack setup
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use \WP_Error;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * General purpose class for managing installation/activation of plugins.
+ */
+class Plugin_Manager {
+
+	/**
+	 * Determine whether plugin installation is allowed in the current environment.
+	 *
+	 * @return bool
+	 */
+	public static function can_install_plugins() {
+		if ( ( defined( 'DISALLOW_FILE_EDIT' ) && DISALLOW_FILE_EDIT ) ||
+			( defined( 'DISALLOW_FILE_MODS' ) && DISALLOW_FILE_MODS ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Activate a plugin, installing it first if necessary.
+	 *
+	 * @param string $plugin The plugin slug or URL to the plugin.
+	 * @return bool True on success. False on failure or if plugin was already activated.
+	 */
+	public static function activate( $plugin ) {
+		if ( ! function_exists( 'get_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_slug = $plugin;
+
+		// if it's a url, parse the slug from the URL.
+		if ( wp_http_validate_url( $plugin ) ) {
```

Comment:
I expected to be able to install a plugin from a .zip file in the filesystem by passing in a local path, e.g. `Plugin_Manager::activate( '/srv/www/wordpress-develop/public_html/build/wp-content/plugins/newspack-plugin/amp-wp-develop.zip' )`, but only an absolute URL (or slug) can be used. I'm not sure we need to support this third approach, but if we do end up with some kind of a Newspack distro this might be a way to improve installation speed by avoiding unnecessary network activity. What do you think?

Reply:
Interesting idea. I would prefer not to have that feature if we don't actively need it, as it introduces a new category of security concerns we need to watch out for. Let's wait and add this feature if/when we need it. :)

Reply:
Agreed—YAGNI